cds-snc/cppd-medical-report-sails
# Add testing framework

# What this does
Adds the minimum required for us to begin testing. Includes cypress-axe and axe-core for a11y testing. Also includes a few basic tests to get us started testing the Patient flow. 

Note that one of the tests fails on FIrefox, something to do with field focus on errors. We can work on this as we build out the test suite.

This does not include any setup for CI, and relies on you having your local environment running to test against. We've gotta start somewhere.

## To test
- Grab this code
- Run `npm i` to install new packages
- Run `npm run cypress:open`
- In the window that opens, run the `patient_flow.spec.js` tests

The test runner will open, and you should see it run through the tests:
- Visit the start page
- Check the heading
- Navigate to the personal page
- Check the heading
- Fill in the SIN field with an invalid sin
- Submit the form
- Check the error message
- Check the field focus (doesn't seem to work on firefox?)
